### PR TITLE
feat: make the special sfx of enemy weapons configurable too

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/zc/weapons.cpp [hunk 1]
```diff
@@ -1191,6 +1193,21 @@ void weapon::eweapon_overrides()
 		}
 	}
 	else step = e->wstep;
+	if (e->firesfx < 0)
```
Comment (connorjclark) [inline src/zc/weapons.cpp]:
```suggestion
	if (e->firesfx <= 0)
```
?

Comment (connorjclark) [reply src/zc/weapons.cpp]: but also, this block of code just seems to repeat what is already happening in the weapon constructor switch, why is it needed?

File: src/zc/guys.cpp [hunk 2]
```diff
@@ -11539,11 +11549,6 @@ void eStalfos::KillWeapon()
 			}
 		}
 	}
-	
-	if(wpn==ewBrang && !Ewpns.idCount(ewBrang))
-	{
-		stop_sfx(WAV_BRANG);
-	}
```
Comment (connorjclark) [inline src/zc/guys.cpp:11546]: wouldn't this keep the sfx playing for a tiny bit even when all dead?

should this be `stop_sfx(specialsfx)` now (though specialsfx is still always the same for any specific enemy class, but I guess you plan to make that configurable)

Comment (ghost) [reply src/zc/guys.cpp:11546]: that codeblock does nothing.

File: src/zc/weapons.cpp (hunk 1 above)
Comment (ghost) [reply src/zc/weapons.cpp]: script created weapons just use the constructor.
enemy created weapons use eweapon_overrides()

File: src/zc/guys.cpp (hunk 2 above)
Comment (connorjclark) [reply src/zc/guys.cpp:11546]: Explain please - is there another part that stops the sfx early when a weapon is deleted?

Comment (ghost) [reply src/zc/guys.cpp:11546]: yes the weapon deconstructor does it

File: src/zc/weapons.cpp (hunk 1 above)
Comment (ghost) [reply src/zc/weapons.cpp]: this is actually incorrect.. should be setting specialsfx to firesfx its been fixed

File: src/base/qst.cpp [hunk 3]
```diff
@@ -9779,14 +9779,20 @@ static void guy_update_firesfx(guydata& tempguy)
 					tempguy.firesfx = WAV_ZN1ICE;
 					break;
 				case ewRock:
-					if (get_qr(qr_MORESOUNDS)) tempguy.firesfx = WAV_ZN1ROCK;
```
Comment (connorjclark) [inline src/base/qst.cpp:9782]: so this is only fine b/c `enemy::wpnsfx` is still checking the qr. hm ok.

**Review by connorjclark — APPROVED** [review]

**Pull request merged**